TESTS/mbed_drivers/rtc/main.cpp

```diff
@@ -16,41 +16,423 @@
+    TEST_ASSERT_EQUAL(CUSTOM_TIME_1, buffer);
+}
+
+/* This test verifies if set_time() function inits
+ * the RTC and writes current time if RTC functions
+ * are defined.
+ *
+ * Note: Stubs are used instead of original RTC functions.
+ *
+ * Given environment has RTC functions defined.
+ * When set_time() functions is called.
+ * Then function initialises RTC and
+ * sets RTC time.
+ */
+void test_set_time_RTC_func_defined()
+{
+    /* Attache RTC read/write/init/isenabled stubs. */
+    attach_rtc(read_rtc_stub, write_rtc_stub, init_rtc_stub, isenabled_rtc_stub);
+
+    /* Set to unexpected. */
+    rtc_time_val = 123;
+    rtc_init_called = false;
+
+    /* Set current time. */
+    rtc_time_val = 123;
+
+    /* Set new RTC time. */
+    set_time(CUSTOM_TIME_1);
+
+    /* Check if RTC init has been performed and RTC time has been updated. */
+    TEST_ASSERT_EQUAL(true, rtc_init_called);
+    TEST_ASSERT_EQUAL(CUSTOM_TIME_1, time(NULL));
+    TEST_ASSERT_EQUAL(CUSTOM_TIME_1, rtc_time_val);
+}
+
+/* This test verifies if set_time() function can be
+ * successfully executed when init RTC function is undefined.
+ *
+ * Note: Stubs are used instead of original RTC functions.
+ *
+ * Given environment has init RTC function undefined.
+ * When set_time() functions is called.
+ * Then function sets RTC time.
+ */
+void test_set_time_init_RTC_func_undefined()
+{
+    /* Attache RTC read/write/isenabled stubs. */
+    attach_rtc(read_rtc_stub, write_rtc_stub, NULL, isenabled_rtc_stub);
+
+    /* Set to unexpected. */
+    rtc_time_val = 123;
+
+    /* Set new RTC time. */
+    set_time(CUSTOM_TIME_1);
+
+    /* Check if RTC time has been updated. */
+    TEST_ASSERT_EQUAL(CUSTOM_TIME_1, time(NULL));
+    TEST_ASSERT_EQUAL(CUSTOM_TIME_1, rtc_time_val);
+}
+
+/* This test verifies if set_time() function can be
+ * successfully executed when write RTC function is undefined.
+ *
+ * Note: Stubs are used instead original RTC functions.
+ *
+ * Given environemt has write RTC function undefined.
+ * When set_time() function is called.
+ * Then function inits RTC and does not modify
+ * RTC time.
+ */
+void test_set_time_write_RTC_func_undefined()
+{
+    /* Attache RTC read/write/init/isenabled stubs. */
+    attach_rtc(read_rtc_stub, NULL, init_rtc_stub, isenabled_rtc_stub);
+
+    /* Set to unexpected. */
+    rtc_time_val = 123;
+    rtc_init_called = false;
+
+    /* Set new RTC time. */
+    set_time(CUSTOM_TIME_1);
+
+    /* Check if RTC has been initialized and RTC time has not been updated. */
+    TEST_ASSERT_EQUAL(true, rtc_init_called);
+    TEST_ASSERT_EQUAL(123, time(NULL));
+    TEST_ASSERT_EQUAL(123, rtc_time_val);
+}
+
+/* This test verifies if RTC time can be successfully
+ * set and read.
+ *
+ * Note: Original RTC functions are used in this test.
+ *
+ * Given environment has RTC available.
+ * When set_time()/time() functions is called.
+ * Then RTC time is set/retrieved.
+ */
+template<uint32_t timeValue>
```

Comment:
I would split that in two tests set and wait.

Reply:
Spitted in to two test cases as suggested.